application/src/main/java/org/thingsboard/server/service/entitiy/ruleChain/DefaultTbRuleChainNotifyService.java

```diff
@@ -0,0 +1,266 @@
+            } else {
+                updatedRuleChains = Collections.emptyList();
+            }
+
+            RuleChainMetaData savedRuleChainMetaData = checkNotNull(ruleChainService.loadRuleChainMetaData(tenantId, ruleChainMetaData.getRuleChainId()));
+
+            if (RuleChainType.CORE.equals(ruleChain.getType())) {
+                tbClusterService.broadcastEntityStateChangeEvent(ruleChain.getTenantId(), ruleChain.getId(), ComponentLifecycleEvent.UPDATED);
+                updatedRuleChains.forEach(updatedRuleChain -> {
+                    tbClusterService.broadcastEntityStateChangeEvent(updatedRuleChain.getTenantId(), updatedRuleChain.getId(), ComponentLifecycleEvent.UPDATED);
+                });
+            }
+
+            notificationEntityService.notifyCreateOrUpdateOrDelete(tenantId, null, ruleChainId,
+                    ruleChain, user, ActionType.UPDATED, false, null, ruleChainMetaData);
+
+            if (RuleChainType.EDGE.equals(ruleChain.getType())) {
+                notificationEntityService.notifySaveRuleChainMetaData(tenantId, ruleChain);
+            }
+
+            for (RuleChain updatedRuleChain : updatedRuleChains) {
+                RuleChainMetaData updatedRuleChainMetaData = checkNotNull(ruleChainService.loadRuleChainMetaData(tenantId,
+                        updatedRuleChain.getId()));
+                notificationEntityService.notifySaveRuleChainMetaData(tenantId, ruleChain, updatedRuleChain,
+                        updatedRuleChainMetaData, user);
+            }
+
+            return savedRuleChainMetaData;
+        } catch (Exception e) {
+            notificationEntityService.notifyCreateOrUpdateOrDelete(tenantId, null, emptyId(EntityType.RULE_CHAIN),
+                    null, user, ActionType.ADDED, false, e,  ruleChainMetaData);
+            throw handleException(e);
+        }
+    }
+
+    @Override
+    public RuleChain assignRuleChainToEdge(TenantId tenantId, RuleChainId ruleChainId, Edge edge, SecurityUser user) throws ThingsboardException {
+        try {
+            RuleChain savedRuleChain = checkNotNull(ruleChainService.assignRuleChainToEdge(tenantId, ruleChainId, edge.getId()));
+            notificationEntityService.notifyAssignOrUnassignEntityToEdge(tenantId, ruleChainId,
+                    null, edge.getId(),
+                    savedRuleChain, ActionType.ASSIGNED_TO_EDGE,
+                    user, ruleChainId.toString(), edge.getId().toString(), edge.getName());
+            return savedRuleChain;
+        } catch (Exception e) {
+            notificationEntityService.notifyCreateOrUpdateOrDelete(tenantId, null, emptyId(EntityType.RULE_CHAIN),
+                    null, user, ActionType.ASSIGNED_TO_EDGE, false, e, ruleChainId.toString(), edge.getId().toString());
+            throw handleException(e);
+        }
+    }
+
+    @Override
+    public RuleChain unassignRuleChainFromEdge(TenantId tenantId, RuleChain ruleChain, Edge edge, SecurityUser user) throws ThingsboardException {
```

Comment:
let's pass **RuleChainId ruleChainId** instead of RuleChain to make it sync with **assignRuleChainToEdge** method

Reply:
Ok